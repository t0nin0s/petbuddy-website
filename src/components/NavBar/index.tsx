import React from 'react';
import styled, { css } from 'styled-components'
import { mediaQueries } from '../../mediaqueries'
import { Link } from 'react-router-dom';
import { NavBarProps } from '../../types'

const NavBarItems = styled.div<NavBarProps>`
display: none;
  ${props => props.primary && css`
    display:flex;
    flex-direction:column;
    padding:1rem;
  `}
  ${mediaQueries("medium")`
    display:flex;
    flex-direction:row;
    justify-content:space-around;
    width:60%;
    align-self:center;
    margin-left: 20rem;
    color:#f48fb1;
    font-weight: 700;
  `}
`
const NavBarLink = styled(Link) <NavBarProps>`

text-decoration:none;
height:2rem;
opacity: 0.6;
transition: 0.3s;
&:hover {
  margin:0;
  opacity:1;
  font-weight:700;
  background: pink;
  padding: 0 .5rem;
}
${mediaQueries('medium')`
  height:auto;
    &:hover {
    width:auto;
    background:whitesmoke;
    transition: 0.5s;
    opacity:1;
    }`
  }`

export default (props: NavBarProps) => {
  const { primary } = props;
  return (
    <NavBarItems primary={primary}>
      <NavBarLink to={'/cat-service'}>Cat Services</NavBarLink>
      <NavBarLink to={'/dog-service'}>Dog Services</NavBarLink>
      <NavBarLink to={'/pet-service'}>Small Pet Services</NavBarLink>
      <NavBarLink to={'/contact'}>Contact Us</NavBarLink>
      <NavBarLink to={'/join'}>Join Us</NavBarLink>
    </NavBarItems>
  )
}

