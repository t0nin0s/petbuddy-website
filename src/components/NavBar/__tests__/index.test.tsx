import React from 'react'
import NavBar from '../index'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('NavBar component', () => {
  it('should render', () => {
    const wrapper = shallow(<NavBar />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
