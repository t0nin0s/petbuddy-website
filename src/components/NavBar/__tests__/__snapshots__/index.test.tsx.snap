// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component should render 1`] = `
<styled.div>
  <Styled(Link)
    to="/cat-service"
  >
    Cat Services
  </Styled(Link)>
  <Styled(Link)
    to="/dog-service"
  >
    Dog Services
  </Styled(Link)>
  <Styled(Link)
    to="/pet-service"
  >
    Small Pet Services
  </Styled(Link)>
  <Styled(Link)
    to="/contact"
  >
    Contact Us
  </Styled(Link)>
  <Styled(Link)
    to="/join"
  >
    Join Us
  </Styled(Link)>
</styled.div>
`;
